standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -3890,9 +3890,20 @@ public CompactionResponse compact2(String dbname, String tableName, String parti
     }
     cr.setType(type);
     cr.setProperties(tblproperties);
+    cr.setInitiatorId(hostname() + "-manual");
+    cr.setInitiatorVersion(HiveMetaStoreClient.class.getPackage().getImplementationVersion());
     return client.compact2(cr);
   }
 
+  private String hostname() {
```

Comment:
i don't think it's the best place for this method.

Reply:
fixed, found a util in metastore, that contains this